src/FFmpegWriter.cpp

```diff
@@ -1042,7 +1042,9 @@ AVStream *FFmpegWriter::add_audio_stream() {
 
 	// Set the sample parameters
 	c->bit_rate = info.audio_bit_rate;
+#if !HAVE_CH_LAYOUT
```

Comment:
So, we no longer need to set `c->channels` manually? I'm assuming this gets set automatically when channel_layout is assigned below? (in the newer code path)

Reply:
Yes ch_layout:

https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html

```
/**
  * An AVChannelLayout holds information about the channel layout of audio data.
  *
  * A channel layout here is defined as a set of channels ordered in a specific
  * way (unless the channel order is AV_CHANNEL_ORDER_UNSPEC, in which case an
  * AVChannelLayout carries only the channel count).
  * All orders may be treated as if they were AV_CHANNEL_ORDER_UNSPEC by
  * ignoring everything but the channel count, as long as av_channel_layout_check()
  * considers they are valid.
  *
  * Unlike most structures in FFmpeg, sizeof(AVChannelLayout) is a part of the
  * public ABI and may be used by the caller. E.g. it may be allocated on stack
  * or embedded in caller-defined structs.
  *
  * AVChannelLayout can be initialized as follows:
  * - default initialization with {0}, followed by setting all used fields
  *   correctly;
  * - by assigning one of the predefined AV_CHANNEL_LAYOUT_* initializers;
  * - with a constructor function, such as av_channel_layout_default(),
  *   av_channel_layout_from_mask() or av_channel_layout_from_string().
  *
  * The channel layout must be unitialized with av_channel_layout_uninit()
  *
  * Copying an AVChannelLayout via assigning is forbidden,
  * av_channel_layout_copy() must be used instead (and its return value should
  * be checked)
  *
  * No new fields may be added to it without a major version bump, except for
  * new elements of the union fitting in sizeof(uint64_t).
  */
 typedef struct [AVChannelLayout](https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html) {
     /**
      * Channel order used in this layout.
      * This is a mandatory field.
      */
     enum [AVChannelOrder](https://ffmpeg.org/doxygen/trunk/group__lavu__audio__channels.html#ga252e1528ae55fbf873266ca4c7e3b69c) [order](https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html#a12d3f7dda8395c51bd5e90dbd948064d);
  
     /**
      * Number of channels in this layout. Mandatory field.
      */
     int [nb_channels](https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html#adfd3f460a8ea1575baa32852d9248d3c);
  
     /**
      * Details about which channels are present in this layout.
      * For AV_CHANNEL_ORDER_UNSPEC, this field is undefined and must not be
      * used.
      */
     union {
         /**
          * This member must be used for AV_CHANNEL_ORDER_NATIVE, and may be used
          * for AV_CHANNEL_ORDER_AMBISONIC to signal non-diegetic channels.
          * It is a bitmask, where the position of each set bit means that the
          * AVChannel with the corresponding value is present.
          *
          * I.e. when (mask & (1 << AV_CHAN_FOO)) is non-zero, then AV_CHAN_FOO
          * is present in the layout. Otherwise it is not present.
          *
          * @note when a channel layout using a bitmask is constructed or
          * modified manually (i.e.  not using any of the av_channel_layout_*
          * functions), the code doing it must ensure that the number of set bits
          * is equal to nb_channels.
          */
         uint64_t [mask](https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html#a78b6087054bf8c308355384d050b5d51);
         /**
          * This member must be used when the channel order is
          * AV_CHANNEL_ORDER_CUSTOM. It is a nb_channels-sized array, with each
          * element signalling the presence of the AVChannel with the
          * corresponding value in map[i].id.
          *
          * I.e. when map[i].id is equal to AV_CHAN_FOO, then AV_CH_FOO is the
          * i-th channel in the audio data.
          *
          * When map[i].id is in the range between AV_CHAN_AMBISONIC_BASE and
          * AV_CHAN_AMBISONIC_END (inclusive), the channel contains an ambisonic
          * component with ACN index (as defined above)
          * n = map[i].id - AV_CHAN_AMBISONIC_BASE.
          *
          * map[i].name may be filled with a 0-terminated string, in which case
          * it will be used for the purpose of identifying the channel with the
          * convenience functions below. Otherise it must be zeroed.
          */
         [AVChannelCustom](https://ffmpeg.org/doxygen/trunk/structAVChannelCustom.html) *[map](https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html#af0e00bccc9a5d934b48c603a9a49c708);
     } [u](https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html#a4a05a3aac57d58ce285c8b536e45d13d);
  
     /**
      * For some private data of the user.
      */
     void *[opaque](https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html#ae4a7b1c772395f34bba287d035ae9653);
 } [AVChannelLayout](https://ffmpeg.org/doxygen/trunk/structAVChannelLayout.html);
```